apps/wpcom-block-editor/src/wpcom/features/live-preview/live-preview-notice-plugin.tsx

```diff
@@ -0,0 +1,86 @@
+import { useDispatch, useSelect } from '@wordpress/data';
+import { __, sprintf } from '@wordpress/i18n';
+import { FC, useEffect } from 'react';
+import { NOTICE_ID } from './constants';
+import { useCanPreviewButNeedUpgrade } from './hooks/use-can-preview-but-need-upgrade';
+import { useHideTemplatePartHint } from './hooks/use-hide-template-part-hint';
+import { usePreviewingTheme } from './hooks/use-previewing-theme';
+import { LivePreviewUpgradeModal } from './upgrade-modal';
+import { LivePreviewUpgradeNotice } from './upgrade-notice';
+import { getUnlock } from './utils';
+
+const unlock = getUnlock();
+
+/**
+ * This is an interim solution to clarify to users that they are currently live previewing a theme.
+ * And this should be moved to jetpack-mu-wpcom.
+ * @see https://github.com/Automattic/wp-calypso/issues/82218
+ */
+const LivePreviewNotice: FC< {
+	dashboardLink?: string;
+	previewingThemeName?: string;
+} > = ( { dashboardLink, previewingThemeName } ) => {
+	const { createInfoNotice, removeNotice } = useDispatch( 'core/notices' );
+
+	useHideTemplatePartHint();
+
+	useEffect( () => {
+		createInfoNotice(
+			sprintf(
+				// translators: %s: theme name
+				__(
+					'You are previewing the %s theme. You can try out your own style customizations, which will only be saved if you activate this theme.',
+					'wpcom-live-preview'
+				),
+				previewingThemeName
+			),
+			{
+				id: NOTICE_ID,
+				isDismissible: false,
+				actions: dashboardLink && [
+					{
+						label: __( 'Back to themes', 'wpcom-live-preview' ),
+						url: dashboardLink,
+						variant: 'secondary',
+					},
+				],
+			}
+		);
+		return () => removeNotice( NOTICE_ID );
+	}, [ dashboardLink, createInfoNotice, removeNotice, previewingThemeName ] );
+	return null;
+};
+
+const LivePreviewNoticePlugin = () => {
+	const siteEditorStore = useSelect( ( select ) => select( 'core/edit-site' ), [] );
+	const previewingTheme = usePreviewingTheme();
+	const { canPreviewButNeedUpgrade, upgradePlan } = useCanPreviewButNeedUpgrade( previewingTheme );
+	const dashboardLink = useSelect(
+		( select ) =>
+			unlock &&
+			select( 'core/edit-site' ) &&
+			unlock( siteEditorStore ).getSettings().__experimentalDashboardLink,
+		[ siteEditorStore ]
+	);
+
+	// Do nothing in the Post Editor context.
+	if ( ! siteEditorStore ) {
```

Comment:
Can we now remove this? We already gate this in LivePreviewPlugin.

Reply:
Sure!